hls-plugin-api/src/Ide/Types.hs

```diff
@@ -776,7 +776,7 @@ instance Semigroup (PluginNotificationHandlers a) where
 instance Monoid (PluginNotificationHandlers a) where
   mempty = PluginNotificationHandlers mempty
 
-type PluginMethodHandler a m = a -> PluginId -> MessageParams m -> LspM Config (Either ResponseError (MessageResult m))
+type PluginMethodHandler a m = a -> PluginId -> MessageParams m -> LspM Config (Either PluginError (MessageResult m))
 
 type PluginNotificationMethodHandler a m = a -> VFS -> PluginId -> MessageParams m -> LspM Config ()
```

Comment:
at the plugin level, I feel like this should be able to return an error also! We won't send it back as a responseError, but we might still want to log it etc.! And surely the notification handlers would also like to use the helpers we're defining that work in `ExceptT`

Reply:
The only plugins that have notification handlers are the cabal plugin and ghcide. They do all of their work in a liftIO block, and don't use any of the wrapped methods we have provided. They also all do their own logging already. Not sure how much we would gain by allowing them to return an error.

Reply:
Hmm okay, I guess they're just a bit weird. But it might be nice to make them less weird later...